src/coreclr/jit/fgdiagnostic.cpp

```diff
@@ -3085,8 +3085,8 @@ void Compiler::fgDebugCheckFlags(GenTree* tree)
         return GenTree::VisitResult::Continue;
     });
 
-    // ADDR nodes break the "parent flags >= operands flags" invariant for GTF_GLOB_REF.
-    if (tree->OperIs(GT_ADDR) && op1->OperIs(GT_LCL_VAR, GT_LCL_FLD))
+    // Addresses of locals never need GTF_GLOB_REF
```

Comment:
Nit: could you keep the mention of the `ADDR` in the comment (here and in morph)?

It makes it stand out as the quirk this special handling is (and will help find places that need to be updated when `ADDR` is deleted).

Reply:
I've readded the check for GT_ADDR, I don't believe we need to handle other cases anyway as they will just be handled when it's their turn in morph anyway.